src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor.Tests/ProcessorFunctionsIntegrationTest.cs

```diff
@@ -32,6 +35,67 @@ public void Dispose()
             Directory.Delete(testInstanceDataFilesDirectory, recursive: true);
         }
     }
+
+    protected void SetupCsvDataFilesForDataSetVersion(ProcessorTestData processorTestData, DataSetVersion dataSetVersion)
+    {
+        var dataSetVersionPathResolver = GetRequiredService<IDataSetVersionPathResolver>();
+
+        var dataSetVersionDir = dataSetVersionPathResolver.DirectoryPath(dataSetVersion);
+        if (!Directory.Exists(dataSetVersionDir))
+        {
+            Directory.CreateDirectory(dataSetVersionDir);
+        }
+
+        // Prepare the data set version directory with data and metadata CSV files
+        File.Copy(sourceFileName: processorTestData.CsvDataGzipFilePath,
+            destFileName: dataSetVersionPathResolver.CsvDataPath(dataSetVersion));
+        File.Copy(sourceFileName: processorTestData.CsvMetadataFilePath,
+            destFileName: dataSetVersionPathResolver.CsvMetadataPath(dataSetVersion));
+    }
+
+    protected async Task<(DataSetVersion dataSetVersion, Guid instanceId)> CreateInitialDataSetVersion(
+        DataSetVersionImportStage importStage,
+        DataSetVersionStatus? status = null,
+        Guid? releaseFileId = null)
+    {
+        DataSet dataSet = DataFixture.DefaultDataSet();
+
+        await AddTestData<PublicDataDbContext>(context => context.DataSets.Add(dataSet));
+
+        DataSetVersionImport dataSetVersionImport = DataFixture
+            .DefaultDataSetVersionImport()
+            .WithStage(importStage);
+
+        DataSetVersion dataSetVersion = DataFixture
+            .DefaultDataSetVersion()
+            .WithDataSet(dataSet)
+            .WithReleaseFileId(releaseFileId ?? Guid.NewGuid())
+            .WithStatus(status ?? DataSetVersionStatus.Processing)
+            .WithImports(() => [dataSetVersionImport])
+            .FinishWith(dsv => dsv.DataSet.LatestDraftVersion = dsv);
+
+        await AddTestData<PublicDataDbContext>(context =>
+        {
+            context.DataSetVersions.Add(dataSetVersion);
+            context.DataSets.Update(dataSet);
+        });
+
+        return (dataSetVersion, dataSetVersionImport.InstanceId);
+    }
+
+    protected void AssertDataSetVersionDirectoryContainsOnlyFiles(
+        DataSetVersion dataSetVersion,
+        params string[] expectedFiles)
+    {
+        var dataSetVersionPathResolver = GetRequiredService<IDataSetVersionPathResolver>();
+        var actualFiles = Directory.GetFiles(dataSetVersionPathResolver.DirectoryPath(dataSetVersion))
+            .Select(Path.GetFileName)
+            .ToArray();
+
+        // Assert that the directory contains the expected files and no others
+        Assert.Equal(expectedFiles.Length, actualFiles.Length);
+        Assert.True(ComparerUtils.SequencesAreEqualIgnoringOrder(expectedFiles, actualFiles));
```

Comment:
In a recent C# version (not sure), the `IEnumerable<T>.Order` method was introduced which means we don't really need this util here. This can be simplified to just:

```cs
Assert.Equal(expectedFiles.Order(), actualFiles.Order());
```

This is a little better for debugging as well because `Assert.True` won't output the enumerables on failure (unlike `Assert.Equal`).

Reply:
Cool! Yeah the debug is much better and the length check that I had prior to it is pretty pointless as well. Thanks for the suggestion.